test/integration/json.js

```diff
@@ -0,0 +1,75 @@
+var _       = require('lodash');
+var Promise = global.testPromise;
+
+
+module.exports = function(bookshelf) {
+
+  var isJsonSupported;
+
+  function checkResponse(actual, expected) {
+    // Knex will store strings if client does not support JSON.
+    if (!isJsonSupported) {
+      expected = _.mapValues(expected, function(value) {
+        if (_.isObject(value)) {
+          return JSON.stringify(value);
+        }
+        return value
+      });
+    }
+    expect(actual).to.eql(expected);
+  }
+
+  before(function() {
+    return require('./helpers/json/supported')(bookshelf).then(function(supported) {
+      isJsonSupported = supported;
+      return require('./helpers/json/migration')(bookshelf)
+    }).then(function() {
+       return require('./helpers/json/inserts')(bookshelf);
+    });
+  });
+
+  describe('JSON support', function() {
+    var Models = require('./helpers/json/objects')(bookshelf).Models;
+
+    var Tank = Models.Tank;
+    var Command = Models.Command;
+
+
+    it('can `fetch` a model with a JSON column', function() {
+      return Command.forge({id: 0}).fetch()
+        .then(function(command) {
+          checkResponse(command.attributes, {
+            id: 0,
+            unit_id: 1,
+            type: 'move',
+            info: {
+              target: {
+                x: 5,
+                y: 10
+              }
+            }
+          });
+        });
+    });
+
+    it('Trying to fetch a model automatically excludes JSON column', function() {
+      if (!checkJson()) return;
```

Comment:
Where is this `checkJson()` coming from? It seems to be causing problems in Travis.